packages/sample/src/main.ts

```diff
@@ -1,23 +1,66 @@
 import { Command } from "commander";
 import { list } from "./command/list.js";
 import { led } from "./command/led.js";
+import {
+    ExpansionHub,
+    openConnectedExpansionHubs,
+    openHubWithAddress,
+    openParentExpansionHub,
+} from "@rev-robotics/expansion-hub";
 
 const program = new Command();
 
 program.version("1.0.0");
 
+program
+    .option("-s --serial <serial>", "serial number")
+    .option("-p --parent <address>", "parent address")
+    .option("-a --address <address>", "module address");
+
 program
     .command("list")
     .description("List all connected expansion hubs")
     .action(async () => {
-        await list();
+        let hubs = await getExpansionHubs();
+        await list(hubs);
     });
 
 program
     .command("led")
     .description("Run LED steps")
     .action(async () => {
-        await led();
+        let hubs = await getExpansionHubs();
+        await led(hubs);
     });
 
 program.parse(process.argv);
+
+async function getExpansionHubs(): Promise<ExpansionHub[]> {
+    let options = program.opts();
+    let serialNumber = options.serial;
+    let moduleAddress = options.address ? Number(options.address) : undefined;
+    let parentAddress = options.parent ? Number(options.parent) : undefined;
```

Comment:
```suggestion
    let moduleAddress = options.address ?? Number(options.address) : undefined;
    let parentAddress = options.parent ?? Number(options.parent) : undefined;
```

Use the [Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) instead, since there are plenty of falsey values that can be passed in. If the user specifies a module address of 0, that should be an error (not that we need to check for that specifically), not ignored.

Reply:
It appears that the options get passed in as strings. Is "0" falsey?

Reply:
This is also not valid nullish coalescing operator syntax. The coalescing operator says "if this value is not null or undefined, return it directly. If it's null or undefined, return this other value".